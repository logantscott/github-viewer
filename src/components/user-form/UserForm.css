.UserForm {
  input {
    padding: 0.5rem;
    margin: 1rem;
  }

  button {
    padding: 0.5rem;
  }
}
