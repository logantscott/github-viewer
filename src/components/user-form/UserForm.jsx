import React from 'react';
import PropTypes from 'prop-types';
import styles from './UserForm.css';

const UserForm = ({ userQuery, onChange, onSubmit }) => (
  <form className={styles.UserForm} onSubmit={onSubmit}>
    <input type="text" value={userQuery} onChange={onChange} />
    <button>Submit</button>
  </form>
);

UserForm.propTypes = {
  userQuery: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default UserForm;
