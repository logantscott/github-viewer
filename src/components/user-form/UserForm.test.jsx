import React from 'react';
import { shallow } from 'enzyme';
import UserForm from './UserForm';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<UserForm userQuery='logantscott' onChange={() => {}} onSubmit={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
