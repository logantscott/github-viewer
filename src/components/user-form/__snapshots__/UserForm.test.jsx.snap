// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<form
  className="UserForm"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="logantscott"
  />
  <button>
    Submit
  </button>
</form>
`;
